crates/frontend/src/api.rs

```diff
@@ -13,12 +14,26 @@ use crate::{
 };
 
 // #[server(GetDimensions, "/fxn", "GetJson")]
-pub async fn fetch_dimensions(tenant: String) -> Result<Vec<Dimension>, ServerFnError> {
+pub async fn fetch_dimensions(
+    filters: ListFilters,
+    tenant: String,
+) -> Result<PaginatedResponse<Dimension>, ServerFnError> {
     let client = reqwest::Client::new();
     let host = use_host_server();
 
-    let url = format!("{}/dimension", host);
-    let response: Vec<Dimension> = client
+    let mut query_params = vec![];
+    if let Some(page) = filters.page {
+        query_params.push(format!("page={}", page));
```

Comment:
dont we have functionality to add query params in url ?

Reply:
Yes we are doing this
checking here as its optional.